domain/network/state/subnet.go

```diff
@@ -80,103 +102,121 @@ VALUES (?, ?)`
-		if _, err := tx.ExecContext(
-			ctx,
-			insertSubnetStmt,
-			uuid.String(),
-			cidr,
-			VLANTag,
-			spaceUUIDValue,
-			subnetType,
-		); err != nil {
-			return errors.Trace(err)
-		}
 
-		if subnetType == subnetTypeFanOverlaySegment {
-			// Retrieve the underlay subnet uuid.
-			var underlaySubnetUUID string
-			row := tx.QueryRowContext(ctx, retrieveUnderlaySubnetUUIDStmt, fanInfo.FanLocalUnderlay)
-			if err := row.Scan(&underlaySubnetUUID); err != nil {
-				return errors.Trace(err)
-			}
-			// Add the association of the underlay and the newly
-			// created subnet to the associations table.
-			if _, err := tx.ExecContext(
-				ctx,
-				insertSubnetAssociationStmt,
-				uuid.String(),
-				underlaySubnetUUID,
-			); err != nil {
-				return errors.Trace(err)
-			}
-		}
-		// Add the subnet uuid to the provider ids table.
-		if _, err := tx.ExecContext(
-			ctx,
-			insertSubnetProviderIDStmt,
-			providerID,
-			uuid.String(),
-		); err != nil {
+	// Add the subnet entity.
+	if _, err := tx.ExecContext(
+		ctx,
+		insertSubnetStmt,
+		subnet.ID,
+		subnet.CIDR,
+		subnet.VLANTag,
+		spaceUUIDValue,
+		subnetType,
+	); err != nil {
+		return errors.Trace(err)
+	}
+
+	if subnetType == subnetTypeFanOverlaySegment {
+		// Retrieve the underlay subnet uuid.
+		var underlaySubnetUUID string
+		row := tx.QueryRowContext(ctx, retrieveUnderlaySubnetUUIDStmt, subnet.FanInfo.FanLocalUnderlay)
+		if err := row.Scan(&underlaySubnetUUID); err != nil {
 			return errors.Trace(err)
 		}
-		// Add the subnet and provider network uuids to the
-		// provider_network_subnet table.
+		// Add the association of the underlay and the newly
+		// created subnet to the associations table.
 		if _, err := tx.ExecContext(
 			ctx,
-			insertSubnetProviderNetworkIDStmt,
-			pnUUID.String(),
-			providerNetworkID,
+			insertSubnetAssociationStmt,
+			subnet.ID,
+			underlaySubnetUUID,
 		); err != nil {
 			return errors.Trace(err)
 		}
-		// Insert the providerNetworkUUID into provider network to
-		// subnets mapping table.
-		if _, err := tx.ExecContext(
-			ctx,
-			insertSubnetProviderNetworkSubnetStmt,
-			pnUUID.String(),
-			uuid.String(),
-		); err != nil {
+	}
+	// Add the subnet uuid to the provider ids table.
+	if _, err := tx.ExecContext(
+		ctx,
+		insertSubnetProviderIDStmt,
+		subnet.ProviderId,
+		subnet.ID,
+	); err != nil {
+		return errors.Trace(err)
+	}
+	// Add the subnet and provider network uuids to the
+	// provider_network_subnet table.
+	if _, err := tx.ExecContext(
+		ctx,
+		insertSubnetProviderNetworkIDStmt,
+		pnUUID.String(),
+		subnet.ProviderNetworkId,
+	); err != nil {
+		return errors.Trace(err)
+	}
+	// Insert the providerNetworkUUID into provider network to
+	// subnets mapping table.
+	if _, err := tx.ExecContext(
+		ctx,
+		insertSubnetProviderNetworkSubnetStmt,
+		pnUUID.String(),
+		subnet.ID,
+	); err != nil {
+		return errors.Trace(err)
+	}
+	for _, az := range subnet.AvailabilityZones {
```

Comment:
Can we break this for loop off from here into a separate method... the addSubnet function is getting rather long.

Reply:
sure, I made a new method only for adding availability zones, should be prettier now